src/averageAges.js

```diff
@@ -14,12 +14,18 @@
  *
  * @return {number}
  */
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+
+  const IsCenturyIncluded = century !== undefined
+    ? men.filter(person => Math.ceil(person.died / 100) === century)
+    : men;
+
+  const result = IsCenturyIncluded.reduce((sum, person) =>
```

Comment:
move to the separate function & reuse in all cases

Reply:
Can't understand what to put in the separate function.
I use everything one time, and as I understood we need functions to avoid duplicating the code

Reply:
So, go to the q&a or tag me in a slack chat